ui/src/dashboards/reducers/cellEditorOverlay.ts

```diff
@@ -157,6 +227,208 @@ export default (state = initialState, action: Action): CEOInitialState => {
+    }
+
+    case ActionType.GroupByTime: {
+      const {queryID, time} = action.payload
+      const nextQueryConfig = groupByTime(getQueryConfig(queryID), time)
+
+      const nextQueries = updateQueryDrafts(nextQueryConfig)
+
+      return {...state, queryDrafts: nextQueries}
+    }
+
+    case ActionType.ToggleTagAcceptance: {
+      const {queryID} = action.payload
+      const nextQueryConfig = toggleTagAcceptance(getQueryConfig(queryID))
+
+      const nextQueries = updateQueryDrafts(nextQueryConfig)
+
+      return {...state, queryDrafts: nextQueries}
+    }
+
+    case ActionType.ToggleField: {
+      const {queryID, fieldFunc} = action.payload
+      const nextQueryConfig = {
+        ...toggleField(getQueryConfig(queryID), fieldFunc),
+        rawText: null,
+      }
+
+      const nextQueries = updateQueryDrafts(nextQueryConfig)
+
+      return {...state, queryDrafts: nextQueries}
+    }
+
+    case ActionType.ApplyFuncsToField: {
+      const {queryID, fieldFunc, groupBy} = action.payload
+      const nextQueryConfig = applyFuncsToField(
+        getQueryConfig(queryID),
+        fieldFunc,
+        groupBy
+      )
+
+      const nextQueries = updateQueryDrafts(nextQueryConfig)
+
+      return {...state, queryDrafts: nextQueries}
+    }
+
+    case ActionType.ChooseTag: {
+      const {queryID, tag} = action.payload
+      const nextQueryConfig = chooseTag(getQueryConfig(queryID), tag)
+
+      const nextQueries = updateQueryDrafts(nextQueryConfig)
+
+      return {...state, queryDrafts: nextQueries}
+    }
+
+    case ActionType.GroupByTag: {
+      const {queryID, tagKey} = action.payload
+      const nextQueryConfig = groupByTag(getQueryConfig(queryID), tagKey)
+
+      const nextQueries = updateQueryDrafts(nextQueryConfig)
+
+      return {...state, queryDrafts: nextQueries}
+    }
+
+    case ActionType.Fill: {
+      const {queryID, value} = action.payload
+      const nextQueryConfig = fill(getQueryConfig(queryID), value)
+
+      const nextQueries = updateQueryDrafts(nextQueryConfig)
+
+      return {...state, queryDrafts: nextQueries}
+    }
+
+    case ActionType.UpdateRawQuery: {
+      const {queryID, text} = action.payload
+      const nextQueryConfig = updateRawQuery(getQueryConfig(queryID), text)
+
+      const nextQueries = updateQueryDrafts(nextQueryConfig)
+
+      return {...state, queryDrafts: nextQueries}
+    }
+
+    case ActionType.EditQueryStatus: {
+      const {queryID, status} = action.payload
+
+      const nextQueryConfig = {...getQueryConfig(queryID), status}
+      const nextQueries = updateQueryDrafts(nextQueryConfig)
+
+      return {...state, queryDrafts: nextQueries}
+    }
+
+    case ActionType.RemoveFuncs: {
+      const {queryID, fields} = action.payload
+      const nextQueryConfig = removeFuncs(getQueryConfig(queryID), fields)
+
+      const nextQueries = updateQueryDrafts(nextQueryConfig)
+
+      // fields with no functions cannot have a group by time
```

Comment:
is this comment misplaced? seems useful but i'm not sure it feels relevant to the line below it